From a496199ca4e89e05579b44402935672e401261b3 Mon Sep 17 00:00:00 2001
From: Ding Wei <leo.ding@rock-chips.com>
Date: Thu, 14 May 2020 09:27:06 +0800
Subject: [PATCH] video: rockchip: mpp: fix issue for px30 switch grf
 workaround

if iommu had paged, then skip the enable process.

Change-Id: I07ae8ca8e0836472f79dcabf7b14f5fbba4b8870
Signed-off-by: Ding Wei <leo.ding@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_iommu.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/video/rockchip/mpp/mpp_iommu.c b/drivers/video/rockchip/mpp/mpp_iommu.c
index f9b8d3d2267d..019d5956e79a 100644
--- a/drivers/video/rockchip/mpp/mpp_iommu.c
+++ b/drivers/video/rockchip/mpp/mpp_iommu.c
@@ -477,11 +477,10 @@ int mpp_iommu_enable(struct mpp_rk_iommu *iommu)
 {
 	int i;
 
-	/* iommu should be paging disable */
-	if (mpp_iommu_is_paged(iommu)) {
-		mpp_err("iommu disable failed\n");
-		return -ENOMEM;
-	}
+	/* check iommu whether is paged */
+	iommu->is_paged = mpp_iommu_is_paged(iommu);
+	if (iommu->is_paged)
+		return 0;
 
 	/* enable stall */
 	for (i = 0; i < iommu->mmu_num; i++)
@@ -520,7 +519,8 @@ int mpp_iommu_enable(struct mpp_rk_iommu *iommu)
 	/* iommu should be paging enable */
 	iommu->is_paged = mpp_iommu_is_paged(iommu);
 	if (!iommu->is_paged) {
-		mpp_err("iommu enable failed\n");
+		mpp_err("iommu->base_addr=%08x enable failed\n",
+			iommu->base_addr[0]);
 		return -EINVAL;
 	}
 
-- 
2.35.3

